lib/daps_functions

```diff
@@ -2056,22 +2060,24 @@ function getimages {
 ###########################################################################
 function spellcheck {
     local SHORT_OPTS LONG_OPTS SUB_CMD
-    local ASPELL_CMD COUNT FILE LISTMODE LISTOUTPUT SPELLCMD
+    local SPELL_CHECKER_CMD COUNT FILE LISTMODE LISTOUTPUT SPELLCMD
     local -a FILELIST SKIPLIST
 
     SUB_CMD=$1
     shift
 
     SHORT_OPTS="h"
-    LONG_OPTS="extra-dict:,file:,help,lang:,list,rootid:"
+    LONG_OPTS="extra-dict:,file:,help,lang:,list,rootid:,spell-checker:"
 
     parse_args "$SHORT_OPTS" "$LONG_OPTS" "$SUB_CMD" "$@"
     eval set -- "$P_REMAIN_ARGS"
 
-    # check whether aspell is installed:
-    ASPELL_CMD="$(which  --skip-alias --skip-functions "aspell" 2>/dev/null)"
-    if [[ -z $ASPELL_CMD ]]; then
-        ccecho "error" "Error: aspell is not installed"
+    [[ -n "$P_SPELL_CHECKER" ]] && SPELL_CHECKER="$P_SPELL_CHECKER"
```

Comment:
To set a default when no spellchecker is specified elsewhere, I suggest the following code (not tested)

        local SC
        ....
        if [[ -n "$P_SPELL_CHECKER" ]]; then
          # checker was provided on the command line
          SPELL_CHECKER="$P_SPELL_CHECKER"
        else
          if [[ -z "$SPELL_CHECKER" ]]; then
           # no spell checker defined in the configs or on the command line
           # searching for aspell or hunspell
            for SC in aspell hunspell; do
              SPELL_CHECKER_CMD="$(which  --skip-alias --skip-functions "$SC" 2>/dev/null)"
              [[ -n "$SPELL_CHECKER_CMD" ]] && break;
            done
            [[ -z "$SPELL_CHECKER_CMD"  && exit_on_error "Cannot perform a spell check; neither aspell nor hunspell is installed"
          fi
        fi

The following makes some of the code below obsolete:

        if [[ -z "$SPELL_CHECKER_CMD" ]]; then
          # the spell checker was not set automatically
          # 
          # check if spell_checker is valid
          [[ "aspell" = $SPELL_CHECKER || "hunspell" = $SPELL_CHECKER ]] ||￼  exit_on_error "Wrong value for --spell_checker. Needs to be \"aspell\" or \"hunspell\""
          # check whether spell is installed:
        ￼  SPELL_CHECKER_CMD="$(which  --skip-alias --skip-functions "$SPELL_CHECKER" 2>/dev/null)"
        ￼  if [[ -z $SPELL_CHECKER_CMD ]]; then
        ￼    ccecho "error" "Error: '$SPELL_CHECKER' is not installed"
        ￼    exit 1
          fi
        fi